gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java

```diff
@@ -91,6 +102,48 @@ public void apply(Project project) {
                                             .setBootClasspath(new LazyConfigurationList(errorProneFiles)));
                         });
             }
+
+            // Create compile*Apply tasks that apply the errorprone-generated patch files
+            Task applyTask = project.getTasks().create(APPLY_TASK_NAME);
+            List<JavaCompile> compileTasks = ImmutableList.copyOf(
+                    project.getTasks().withType(JavaCompile.class).iterator());
+            compileTasks.forEach(javaCompile -> {
+                File patchFile = project.getBuildDir().toPath()
+                        .resolve("errorprone")
+                        .resolve(javaCompile.getName())
+                        .resolve("error-prone.patch")
+                        .toFile();
+                javaCompile.getOutputs().file(patchFile);
+
+                // TODO(gatesn): How can we flag this on/off? It appears to come with a hefty perf hit
```

Comment:
in tritium i use https://github.com/palantir/tritium/blob/develop/build.gradle#L110 and https://github.com/palantir/tritium/blob/develop/gradle.properties#L4 then when I want to generate a patch one can `./gradlew -DgenerateErrorPronePatch=true compileJava compileTestJava`